superset/datasets/api.py

```diff
@@ -330,7 +330,6 @@ def post(self) -> Response:
 
     @expose("/<pk>", methods=("PUT",))
     @protect()
-    @safe
```

Comment:
@betodealmeida could you remind us what this decorator did?

Reply:
It's from FAB, it catches unhandled exceptions and returns a 500 JSON payload. The problem is that the payload is not [SIP-40](https://github.com/apache/superset/issues/9194) compatible. When we remove the decorator [Flask will catch any exceptions and return a SIP-40 compatible JSON payload](https://github.com/apache/superset/blob/ec6191023263ac5f8292dd37f037805c3196e029/superset/views/base.py#L519-L535), so it's not needed at all.